src/core/shuntingYard.hpp

```diff
@@ -380,7 +380,13 @@ std::string commands(std::string & expr, bool quotes = true) {
             JsonObject   output = doc_out.to<JsonObject>();
             JsonObject   input  = doc_in.to<JsonObject>();
             std::string  cmd_s  = "api/" + std::string(cmd);
-            emsesp::Command::process(cmd_s.c_str(), true, input, output);
+
+            auto return_code = emsesp::Command::process(cmd_s.c_str(), true, input, output);
+            // check for no value (entity is valid but has no value set)
+            if (return_code == emsesp::CommandRet::NO_VALUE) {
+                return expr = ""; // just ignore for now
+            }
+
```

Comment:
I think this is not needed, it returns an empy string that we can insert here as `""`, should work with a compare like `boiler/rettemp==""` (to test if a sensor is connected)

Reply:
Oh, empty strings crashes the shuntingYard, i think it's the s[0] access in isnum(), maybe also others. Will check.

Reply:
if could check that would be good. I'll also add a test and try and reproduce (using pio run -e native-test -t exec)

Reply:
This change work with empty strings:
https://github.com/MichaelDvP/EMS-ESP32/commit/071e81f29bbdb21d274a79d5aaf8849c5242d233
Tested with `boiler/rettemp == ""` (returns `1`, my boiler does not have a sensor)

Reply:
shall I merge into my PR? Or you overlay on after I merge?

Reply:
I'll PR now